yt/visualization/plot_modifications.py

```diff
@@ -733,10 +734,30 @@ def __call__(self, plot):
         nx = plot.raw_image_shape[1] // self.factor[0]
         ny = plot.raw_image_shape[0] // self.factor[1]
         xx0, xx1, yy0, yy1 = self._plot_bounds(plot)
-        X, Y = np.meshgrid(
-            np.linspace(xx0, xx1, nx, endpoint=True),
-            np.linspace(yy0, yy1, ny, endpoint=True),
-        )
+
+        if plot._transform is None:
+            X, Y = np.meshgrid(
+                np.linspace(xx0, xx1, nx, endpoint=True),
+                np.linspace(yy0, yy1, ny, endpoint=True),
+            )
+        else:
+            # when we have a cartopy transform, provide the x, y values
+            # in the coordinate reference system of the data and let cartopy
+            # do the transformation. Also check for the exact bounds of the transform
+            # which can cause issues with projections.
+            if _check_geographic_bounds(bounds, plot._transform):
+                # note: cartopy will also raise its own warning, but it is useful to add this
+                # warning as well since the only way to avoid the exact bounds is to change the
+                # extent of the plot.
```

Comment:
note to reviewers: I initially pushed a version that offset the bounds by a small amount to avoid the exact geographic bounds of the transform but in the end decided it was better to just warn the user and let them change the plot width if they want to silence the warning. I could go back to the offset approach, but there might be some annoying edge cases to worry about in deciding what value to use for the offset.